packages/grafana-data/src/types/datasource.ts

```diff
@@ -204,6 +205,9 @@ abstract class DataSourceApi<
     this.type = instanceSettings.type;
     this.meta = {} as DataSourcePluginMeta;
     this.uid = instanceSettings.uid;
+    if (!this.uid) {
+      this.uid = this.name;
+    }
```

Comment:
what is this condition for? Some of the built in (-- Dashboard --, -- Grafana --) data sources?

Reply:
yes the builtin ones do not have uid (or id)